Recently added assert in LogToXmlHandler.py is tripping up Python logging's shutdown()

On logging shutdown, which can happen in an atexit handler after any Arelle Cntlr has gone away, I'm getting the below traceback since https://github.com/Arelle/Arelle/commit/2c8bca40583d261e40eeb038cf04913bf2c533e6 was added:

```python
Exception ignored in atexit callback <function shutdown at 0x00000224F288A770>:
Traceback (most recent call last):
  File "C:\Users\stuar\AppData\Local\Python\pythoncore-3.14-64\Lib\logging\__init__.py", line 2245, in shutdown
    h.flush()
  File "C:\Users\stuar\Documents\efrag\Digital-Template-to-XBRL-Converter\.venv-py314\Lib\site-packages\arelle\logging\handlers\LogToXmlHandler.py", line 48, in flush
    assert self.cntlr is not None, \
AssertionError: Controller object is required for security plugin hooks to execute.
```

I think the assert and plugin block wants to change to something like:
```python
securityIsActive = securityHasWritten = False
if self.cntlr is not None:
    for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.IsActive"):
        securityIsActive = pluginMethod(self) # must be active for the cntlr object to effect log writing
```

--- a/arelle/logging/handlers/LogToXmlHandler.py
+++ b/arelle/logging/handlers/LogToXmlHandler.py
@@ -6,8 +6,10 @@ from __future__ import annotations
 import json
 import logging
 import sys
-from typing import TYPE_CHECKING
 
+from typing import TYPE_CHECKING, Iterator, Callable, Any
+
+from arelle import PluginManager
 from arelle.logging.handlers.LogHandlerWithXml import LogHandlerWithXml
 from arelle.typing import TypeGetText
 
@@ -42,12 +44,16 @@ class LogToXmlHandler(LogHandlerWithXml):
         self.logRecordBuffer = []
         self.filemode = mode
 
+    def _hooks(self, name: str) -> Iterator[Callable[..., Any]]:
+        if self.cntlr is not None:
+            yield from self.cntlr.plugins.hooks(name)
+        else:
+            yield from PluginManager.getInstance().pluginClassMethods(name)
+
     def flush(self) -> None:
         # Note to developers: breakpoints in this method don't work, please debug with print statements
         securityIsActive = securityHasWritten = False
-        assert self.cntlr is not None, \
-            _("Controller object is required for security plugin hooks to execute.")
-        for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.IsActive"):
+        for pluginMethod in self._hooks("Security.Crypt.IsActive"):
             securityIsActive = pluginMethod(self) # must be active for the cntlr object to effect log writing
         if self.filename == "logToStdOut.xml":
             print('<?xml version="1.0" encoding="utf-8"?>')
@@ -66,7 +72,7 @@ class LogToXmlHandler(LogHandlerWithXml):
             if self.filename.endswith(".xml"):
                 # print ("filename=" + self.filename)
                 if securityIsActive:
-                    for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.Write"):
+                    for pluginMethod in self._hooks("Security.Crypt.Write"):
                         securityHasWritten = pluginMethod(self, self.filename,
                                                           '<?xml version="1.0" encoding="utf-8"?>\n<log>\n' +
                                                           ''.join(self.recordToXml(logRec) for logRec in self.logRecordBuffer) +
@@ -79,14 +85,14 @@ class LogToXmlHandler(LogHandlerWithXml):
                         fh.write('</log>\n')
             elif self.filename.endswith(".json"):
                 if securityIsActive:
-                    for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.Write"):
+                    for pluginMethod in self._hooks("Security.Crypt.Write"):
                         securityHasWritten = pluginMethod(self, self.filename, self.getJson())
                 if not securityHasWritten:
                     with open(self.filename, self.filemode, encoding='utf-8') as fh:
                         fh.write(self.getJson())
             elif self.filename.endswith(".html"):
                 if securityIsActive:
-                    for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.Write"):
+                    for pluginMethod in self._hooks("Security.Crypt.Write"):
                         securityHasWritten = pluginMethod(self, self.filename, self.getHtml())
                 if not securityHasWritten:
                     with open(self.filename, self.filemode, encoding='utf-8') as fh:
@@ -105,7 +111,7 @@ class LogToXmlHandler(LogHandlerWithXml):
                               file=_file)
             else:
                 if securityIsActive:
-                    for pluginMethod in self.cntlr.plugins.hooks("Security.Crypt.Write"):
+                    for pluginMethod in self._hooks("Security.Crypt.Write"):
                         securityHasWritten = pluginMethod(self, self.filename,
                                                           ''.join(self.format(logRec) + "\n" for logRec in self.logRecordBuffer))
                 if not securityHasWritten:

